src/Component/src/Metadata/Operation/HttpOperationInitiator.php

```diff
@@ -16,13 +16,15 @@
 use Sylius\Resource\Metadata\HttpOperation;
 use Sylius\Resource\Metadata\RegistryInterface;
 use Sylius\Resource\Metadata\Resource\Factory\ResourceMetadataCollectionFactoryInterface;
+use Sylius\Resource\Symfony\ExpressionLanguage\VarsResolverInterface;
 use Symfony\Component\HttpFoundation\Request;
 
 final class HttpOperationInitiator implements HttpOperationInitiatorInterface
 {
     public function __construct(
         private RegistryInterface $resourceRegistry,
         private ResourceMetadataCollectionFactoryInterface $resourceMetadataCollectionFactory,
+        private ?VarsResolverInterface $varsResolver = null,
```

Comment:
Does this place need a deprecation notice, that not providing VarsResolver is not supported anymore and will be required at some future version?